src/sitemapRoutesHandler.js

```diff
@@ -0,0 +1,51 @@
+const fetch = require("isomorphic-fetch");
+const config = require("./config");
+const logger = require("./lib/logger");
+
+/**
+ * @summary processes requests for sitemaps
+ * @param {Object} req - the request object
+ * @param {Object} res - the response object
+ * @param {Object} next - the next middleware function
+ * @returns {undefined}
+ */
+function sitemapRoutesHandler(req, res, next) {
+  if (req.originalUrl.startsWith("/sitemap") === false) {
+    return next();
+  }
+
+  res.setHeader("Content-Type", "text/xml");
+
+  const shopUrl = `${req.protocol}://${req.get("host")}`;
+  const handle = req.originalUrl.replace("/", "");
+
+  const query = `
+    { sitemap(handle: "${handle}", shopUrl: "${shopUrl}") {
+      shopId
+      xml
+    } }
+  `;
+
+  return fetch(config.INTERNAL_GRAPHQL_URL, {
```

Comment:
Worth flagging here. Should this be `INTERNAL_GRAPHQL_URL` or `EXTERNAL_GRAPHQL_URL`? I made the decision based on the fact that the internal is the one added for nextjs `serverRuntimeConfig`; and I'm using it on the server.

Reply:
@aldeed ^^ see question above

Reply:
Should be correct. Internal should be used whenever it can be, external only when it's not on the Docker network, such as making a request from the browser.